File: java dsa codes/stacklinkedlist.java
public class stacklinkedlist {
    static class node{
        int data;
        node next;
    }

    public static int isempty(node top){
        if (top==null){
            return 1;
        }
        else{
            return 0;
        }
    }

    public static int push(node top,int val){
        node p=new node();
        p.data=val;
        p.next=top;
        top=p;
        return 0;
    }

    public static int pop(node top){
        if(isempty(top)==1){
            System.out.println("stack undeflow");
            return -1;
        }
        else{
            int val=top.data;
            top=top.next;
            System.out.println("the value is popped"+val);
            return val;
        }
    }

    public static int isfull(node top){
        node p=new node();
        if(p==null){
            return 1;
        }
        else{
            return 0;
        }
    }

    public static int peek(node top,int pos){
        node p=top;
        for(int i=0;i<pos-1 && p!=null;i++){
            p=p.next;
        }
        if(p!=null){
            return p.data;
        }
        else{
            return -1;
        }
    }

    public static void main(String [] args){
        node top=new node();
        top.data=10;
        top.next=null;
    }    
}

File: c dsa codes/stacklinkedlist.c
#include<stdio.h>
#include<stdlib.h>
struct node{
    int data;
    struct node* next;
};
int push(struct node * top,int data){
    if(isfull(top)){
        printf("stack overflow");
    }
    else{
        struct node* n=(struct node*)malloc(sizeof(struct node));
        n->data=data;
        n->next=top;
        top=n;
        return top;
    }
}

int isempty(struct node * top){
    if(top==NULL){
        return 1;
    }
    else{
        return 0;
    }    
}
int pop(struct node* top){
    if(isempty(top)){
        printf("stack underflow");
    }
    else{
        struct node * n=top;
        top=top->next;
        int popped=n->data;
        free(n);
        return popped;
    }
}

int isfull(struct node * top){
    struct node *b=(struct node*)malloc(sizeof(struct node));
    if(b==NULL){
        return 1;
    }
    else{
        return 0;
    }
}

void print(struct node* top){
    struct node* ptr=top;
    while(ptr!=NULL){
        printf("Element: %d\n",ptr->data);
        ptr=ptr->next;
    }
}

int peek(int pos, struct node* top){
    struct node* ptr=top;
    for(int i=0;(i<pos-1 && ptr!=NULL);i++){
        ptr=ptr->next;
    }
    if(ptr!=NULL){
        return ptr->data;
    }
    else{
        return -1;
    }
}

int stacktop(struct node* top){
    if(isempty(top)){
        printf("stack is empty");
        return -1;
    }
    else{
        return top->data;
    }
}

int stackbottom(struct node* top){
    struct node * p=top;
    while(p->next!=NULL){
        p=p->next;
    }
    return p->data;
}

int main(){
    struct node *top=(struct node*)malloc(sizeof(struct node));
    top->data=10;
    top->next=NULL;
    top=push(top,20);
    
    top=push(top,30);
    print(top);
    
    return 0;
}